test/perf/benchmarks/Engine.ScriptBlock.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+using System.Runtime.InteropServices;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace Engine
+{
+    [BenchmarkCategory(Categories.Engine, Categories.Public)]
+    public class Scripting
+    {
+        private Runspace runspace;
+        private ScriptBlock scriptBlock;
+
+        private void SetupRunspace()
+        {
+            runspace = RunspaceFactory.CreateRunspace(InitialSessionState.CreateDefault2());
+            runspace.Open();
+            Runspace.DefaultRunspace = runspace;
+        }
+
+        #region Invoke-Method
+
+        [ParamsSource(nameof(ValuesForScript))]
+        public string InvokeMethodScript { get; set; }
+
+        public IEnumerable<string> ValuesForScript()
+        {
+            yield return @"'String'.GetType()";
+            yield return @"[System.IO.Path]::HasExtension('')";
+
+            // Test on COM method invocation.
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                yield return @"$sh=New-Object -ComObject Shell.Application; $sh.Namespace('c:\')";
+                yield return @"$fs=New-Object -ComObject scripting.filesystemobject; $fs.Drives";
+            }
+        }
+
+        [GlobalSetup(Target = nameof(InvokeMethod))]
+        public void GlobalSetup()
+        {
+            SetupRunspace();
```

Comment:
Should we make sure that the PowerShell / .NET telemetry is off.

Reply:
Good question. Maybe we should disable it and I will do that from the command line level.